src/redemptions/redemption.service.ts

```diff
@@ -246,6 +255,34 @@ export class RedemptionService {
     return { eligible: true, reason: '' };
   }
 
+  currentDate = (): Date => {
```

Comment:
Is the only reason we have this to mock out in the tests? There should be a way to mock the time via Jest or the Date constructor without needing this.

Reply:
Seems like it could be flakey though if there is a different Date constructor call that is introduced in future. Also could be flakey if we mock all dates and some other check relies on date constructor.

Reply:
If something is based on the current live time, then I think it's reasonable to mock the system time via Jest. All Date constructors then should be consistent

Reply:
for reference I tried:
`jest.useFakeTimers().setSystemTime(new Date('2090-03-06'));`

Reply:
```suggestion
  currentDate(): Date {
```

Discussed offline, but it's behaving a bit weird. I prefer just mocking the `Date` constructor since I think that's cleaner than having a `currentDate` on a service